Refactor database

cyyynthia opened the pull request:
In fact it's more of a general rewrite but yes

cyyynthia committed:
Minor cleanup

cyyynthia committed:
Merge remote-tracking branch 'origin/master'

cyyynthia committed:
Y I K E S

cyyynthia committed:
Database memes and some traces of redis

cyyynthia committed:
Enterprisify remaining entities

cyyynthia committed:
Add all collections

cyyynthia committed:
Massive cleanup of the upload manager

cyyynthia committed:
apparently implicit conversions makes your shit enterprisegarbage

cyyynthia committed:
Massive entity cleanup

And make the thing compile but thats details

cyyynthia committed:
Late night shitcode, probably low quality

cyyynthia committed:
Asyncify collections and some Redis stuff

I swear to god this is GARBAGE

cyyynthia committed:
Rewrite register/login and some other stuff

cyyynthia committed:
Boomerware is no more

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuildChannels.java:
```diff
@@ -28,137 +28,76 @@
-            return;
+    @Override
+    @SuppressWarnings("DuplicateBranchesInSwitch")
+    protected boolean hasPermission(RoutingContext ctx, CrudContext context) {
```
```suggestion
    protected boolean hasPermission(final RoutingContext ctx, final CrudContext context) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuildChannels.java:
```diff
@@ -28,137 +28,76 @@
+    }
 
+    @Override
+    protected IChannel createEntity(RoutingContext ctx) {
```
```suggestion
    protected IChannel createEntity(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuildChannels.java:
```diff
@@ -28,137 +28,76 @@
-
-
+    @Override
+    protected Bson composeUpdate(RoutingContext ctx) {
```
```suggestion
    protected Bson composeUpdate(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/channels/ModuleMessages.java:
```diff
@@ -27,27 +27,70 @@
+
+    @Override
+    @SuppressWarnings("DuplicateBranchesInSwitch")
+    protected boolean hasPermission(RoutingContext ctx, CrudContext context) {
```
```suggestion
    protected boolean hasPermission(final RoutingContext ctx, final CrudContext context) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/channels/ModuleMessages.java:
```diff
@@ -27,27 +27,70 @@
 
-    private void sendMessage(final RoutingContext ctx) { // TODO
+    @Override
+    protected IMessage createEntity(RoutingContext ctx) {
```
```suggestion
    protected IMessage createEntity(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/channels/ModuleMessages.java:
```diff
@@ -27,27 +27,70 @@
+    }
+
+    @Override
+    protected Bson composeUpdate(RoutingContext ctx) {
```
```suggestion
    protected Bson composeUpdate(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/scheduling/SchedulerManager.java:
```diff
@@ -34,7 +34,7 @@
 public class SchedulerManager implements Runnable {
     private static final Logger LOG = LoggerFactory.getLogger(SchedulerManager.class);
-    private LinkedList<ITask<?>> fifo = new LinkedList<ITask<?>>();
+    private LinkedList<ITask<?>> fifo = new LinkedList<>();
```
```suggestion
    private final LinkedList<ITask<?>> fifo = new LinkedList<>();
```

vinceh121 commented on src/main/java/chat/squirrel/scheduling/SchedulerManager.java:
```diff
@@ -34,7 +34,7 @@
     private static final Logger LOG = LoggerFactory.getLogger(SchedulerManager.class);
-    private LinkedList<ITask<?>> fifo = new LinkedList<ITask<?>>();
+    private LinkedList<ITask<?>> fifo = new LinkedList<>();
     private Thread executorThread;
```
```suggestion
    private final Thread executorThread;
```

vinceh121 commented on src/main/java/chat/squirrel/scheduling/SchedulerManager.java:
```diff
@@ -53,7 +53,7 @@ public void shutdown() {
 
     public void addTask(ITask<?> task) {
```
```suggestion
    public void addTask(final ITask<?> task) {
```

vinceh121 commented on src/main/java/chat/squirrel/scheduling/SchedulerManager.java:
```diff
@@ -85,7 +86,7 @@ public void run() {
 
             }
         } catch (InterruptedException e) {
```
```suggestion
        } catch (final InterruptedException e) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/IUser.java:
```diff
@@ -24,80 +24,142 @@
     }
 
-    String getCustomEmail();
+    String getUsername();
```
```suggestion
    @Override
    String getUsername();
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/IUser.java:
```diff
@@ -24,80 +24,142 @@
-    Asset getAvatar();
+    void setUsername(String username);
 
     int getDiscriminator();
```
```suggestion
    @Override
    int getDiscriminator();
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/IUser.java:
```diff
@@ -24,80 +24,142 @@
+     *
+     * @return The flags integer of this user
+     */
     int getFlags();
```
```suggestion
    @Override
    int getFlags();
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/IUser.java:
```diff
@@ -24,80 +24,142 @@
+
+    void setEmail(String email);
+
+    String getAvatar();
```
```suggestion
    @Override
    String getAvatar();
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/impl/RoleImpl.java:
```diff
@@ -25,64 +25,58 @@
     @Override
-    public String getName() {
-        return this.name;
+    public void setGuildId(ObjectId guildId) {
```
```suggestion
    public void setGuildId(final ObjectId guildId) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuilds.java:
```diff
@@ -28,54 +28,82 @@
-            this.fail(ctx, 400, "Invalid JSON payload", null);
-            return;
+    @Override
+    protected boolean hasPermission(RoutingContext ctx, AbstractCrudModule.CrudContext context) {
```
```suggestion
    protected boolean hasPermission(final RoutingContext ctx, final AbstractCrudModule.CrudContext context) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuilds.java:
```diff
@@ -28,54 +28,82 @@
-            this.fail(ctx, 400, "Missing fields", null);
-            return;
+    @Override
+    protected IGuild createEntity(RoutingContext ctx) {
```
```suggestion
    protected IGuild createEntity(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuilds.java:
```diff
@@ -28,54 +28,82 @@
 
-        Squirrel.getInstance().getDatabaseManager().insertEntity(SquirrelCollection.GUILDS, newGuild);
+    @Override
+    protected CompletionStage<InsertOneResult> insertEntity(IGuild entity) {
```
```suggestion
    protected CompletionStage<InsertOneResult> insertEntity(final IGuild entity) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/impl/UserImpl.java:
```diff
@@ -183,54 +234,84 @@ public void setIps(final Collection<String> ips) {
+    }
+
+    @Override
+    public void setDeletionScheduledAt(Date deletionScheduledAt) {
```
```suggestion
    public void setDeletionScheduledAt(final Date deletionScheduledAt) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/FileType.java:
```diff
@@ -0,0 +1,63 @@
+    private final byte[] magicBytes;
+    private final String extension;
+
+    FileType(byte[] magicBytes, String extension) {
```
```suggestion
    FileType(final byte[] magicBytes, final String extension) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/FileType.java:
```diff
@@ -0,0 +1,63 @@
+    // Video
+    MP4(null, "mp4");
+
+    public static FileType[] images = {JPG, PNG, WEBP, GIF, APNG};
```
Might need to make those dynamic

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return input;
+    }
+
+    public void setInput(InputStream input) {
```
```suggestion
    public void setInput(final InputStream input) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return filename;
+    }
+
+    public void setFilename(String filename) {
```
```suggestion
    public void setFilename(final String filename) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return bucket;
+    }
+
+    public void setBucket(Bucket bucket) {
```
```suggestion
    public void setBucket(final Bucket bucket) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return resourceId;
+    }
+
+    public void setResourceId(ObjectId resourceId) {
```
```suggestion
    public void setResourceId(final ObjectId resourceId) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return maxFileSize;
+    }
+
+    public void setMaxFileSize(int maxFileSize) {
```
```suggestion
    public void setMaxFileSize(final int maxFileSize) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return allowedTypes;
+    }
+
+    public void setAllowedTypes(FileType... allowedTypes) {
```
```suggestion
    public void setAllowedTypes(final FileType... allowedTypes) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return allowAnimated;
+    }
+
+    public void setAllowAnimated(boolean allowAnimated) {
```
```suggestion
    public void setAllowAnimated(final boolean allowAnimated) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return resizeWidth;
+    }
+
+    public void setResizeWidth(int resizeWidth) {
```
```suggestion
    public void setResizeWidth(final int resizeWidth) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/UploadAction.java:
```diff
@@ -0,0 +1,126 @@
+        return ratio;
+    }
+
+    public void setRatio(int w, int h) {
```
```suggestion
    public void setRatio(final int w, final int h) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/managers/LocalUploadManager.java:
```diff
@@ -27,184 +27,228 @@
-            } catch (final IOException e1) {
-                res.setErrorReason("Could not write to output stream or digest: " + e1.toString());
+                image = ImageIO.read(action.getInput());
+            } catch (IOException e) {
```
```suggestion
            } catch (final IOException e) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/managers/LocalUploadManager.java:
```diff
@@ -27,184 +27,228 @@
-            } catch (final IOException e) {
-                res.setErrorReason("Failed to close output stream: " + e.toString());
+                ImageIO.write(image, "png", new File(folder, id + ".png"));
+            } catch (IOException e) {
```
```suggestion
            } catch (final IOException e) {
```

vinceh121 commented on src/main/java/chat/squirrel/upload/managers/LocalUploadManager.java:
```diff
@@ -27,184 +27,228 @@
+        if (file.isDirectory()) {
+            final File[] files = file.listFiles();
+            if (files != null) {
+                for (File f : files) {
```
```suggestion
                for (final File f : files) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/AbstractCrudChildEntity.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    @Override
+    protected void registerCrud(String route) {
```
```suggestion
    protected void registerCrud(final String route) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuildRoles.java:
```diff
@@ -27,16 +27,44 @@
+    }
+
+    @Override
+    protected boolean hasPermission(RoutingContext ctx, CrudContext context) {
```
```suggestion
    protected boolean hasPermission(final RoutingContext ctx, final CrudContext context) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuildRoles.java:
```diff
@@ -27,16 +27,44 @@
+    }
+
+    @Override
+    protected IRole createEntity(RoutingContext ctx) {
```
```suggestion
    protected IRole createEntity(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/modules/guilds/ModuleGuildRoles.java:
```diff
@@ -27,16 +27,44 @@
+    }
+
+    @Override
+    protected Bson composeUpdate(RoutingContext ctx) {
```
```suggestion
    protected Bson composeUpdate(final RoutingContext ctx) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/memory/ClassicMemoryAdapter.java:
```diff
@@ -0,0 +1,62 @@
+import java.util.concurrent.CompletionStage;
+
+// TODO
+public class ClassicMemoryAdapter implements IMemoryAdapter { // Aww memowwy NOM NOM NOM NOOOOOOOM
```
BasicMemoryAdapter

vinceh121 commented on src/main/java/chat/squirrel/database/memory/ClassicMemoryAdapter.java:
```diff
@@ -0,0 +1,62 @@
+// TODO
+public class ClassicMemoryAdapter implements IMemoryAdapter { // Aww memowwy NOM NOM NOM NOOOOOOOM
+    @Override
+    public CompletionStage<BsonDocument> getEntity(String key) {
```
```suggestion
    public CompletionStage<BsonDocument> getEntity(final String key) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/memory/ClassicMemoryAdapter.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Override
+    public CompletionStage<List<KeyValue<String, BsonDocument>>> getEntities(String... keys) {
```
```suggestion
    public CompletionStage<List<KeyValue<String, BsonDocument>>> getEntities(final String... keys) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/memory/ClassicMemoryAdapter.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Override
+    public CompletionStage<Boolean> setEntity(String key, BsonDocument document) {
```
```suggestion
    public CompletionStage<Boolean> setEntity(final String key, final BsonDocument document) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/memory/ClassicMemoryAdapter.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Override
+    public CompletionStage<Long> deleteOne(String key) {
```
```suggestion
    public CompletionStage<Long> deleteOne(final String key) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/memory/ClassicMemoryAdapter.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Override
+    public CompletionStage<Long> deleteMany(String... keys) {
```
```suggestion
    public CompletionStage<Long> deleteMany(final String... keys) {
```

vinceh121 commented on src/main/java/chat/squirrel/database/entities/impl/AuditImpl.java:
```diff
@@ -25,45 +25,70 @@
+    private ObjectId guild, user;
+    private AuditLogEntryType type;
+    private String customType;
+    private Map<String, Object> data;
```
Maybe have generic interface to prevent insertion of garbage

vinceh121 reviewed (approved):
Seems alright

cyyynthia committed:
Apply suggestions from ~~meme~~ code review

final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final final

Co-Authored-By: Vincent <vincent.hyvert@sfr.fr>
